111-use-buildah-instead-of-kaniko.md

```diff
@@ -0,0 +1,236 @@
+However, the Kaniko project was [archived](https://github.com/GoogleContainerTools/kaniko/commit/236ba5690eda9170d0157aa8137ebbeb09d38685) and it's no longer developed or maintained.
+This proposal aims to resolve this issue.
+
+## Motivation
+
+The main motivation behind this proposal is to resolve the issue that Kaniko is archived, and it will not receive any updates (no bug or CVE fixes).
+Currently, it seems to not be a problem, but when some critical CVE or bug arises, we have to fix it.
+Also, we need a support in case that there will be something blocking us on newer versions of Kubernetes, which we would not get in this case anymore.
+
+## Proposal
+
+For the Connect Build feature on Kubernetes, we will use [Buildah](https://buildah.io/) as a replacement of Kaniko.
+Buildah is well-supported, widely used tool for building container images.
+It also appears that OpenShift’s Build API relies on Buildah, which makes Buildah a strong alternative to Kaniko, with the added assurance of long-term support and maintenance.
+Other than that, Buildah doesn't need any workaround in order to run it on Kubernetes rootlessly - the only thing that is needed during the build and push stages is to specify `--storage-driver=vfs` option.
+VFS (Virtual File System) driver ensures user-space implementation, it stores each layer as a full copy of the files, but without need of root permissions.
+The only trade-off is the build time (it's slower) and consumption of disk space.
+
+Even though we will run Buildah with these options, which should make it rootless, it still needs some capabilities to run without issues during the build phase.
+That means we cannot run the build Pod with restricted Pod Security profile, as these capabilities are dropped in these profiles.
+This is more described in the [Running Buildah in restricted environment](#running-buildah-in-restricted-environment) section of this proposal.
+
+We will use the official Buildah images on Quay.io - currently `quay.io/buildah/stable:v1.40.1` - and similarly to Kaniko executor image - we will pull the official image, tag it with the version and push it to the Strimzi repository on Quay.
+That's useful because of our versioning and we will have the image ready even if someone on Buildah side decides to remove it from their Quay repository.
+The full name of our Buildah image will be `quay.io/strimzi/buildah:OPERATOR_VERSION` - for `main` branch it will use the `latest` tag, otherwise it will be tagged by the operator version for the particular release.
+
+Users then can specify their own Buildah image using `STRIMZI_DEFAULT_BUILDAH_IMAGE` (similarly to what is possible today with Kaniko).
+In case that users will use Kaniko, the Buildah environment variable will be ignored, and vice versa.
+The users will be responsible for changing these environment variables based on the mode in which they run the Connect build - in case that they have customized Deployment files.
```

Comment:
Is this the cause of the misunderstanding in https://github.com/strimzi/proposals/pull/170#discussion_r2347247825? I'm not sure how exactly you meant it. But both `STRIMZI_DEFAULT_BUILDAH_IMAGE` and `STRIMZI_DEFAULT_KANIKO_EXECUTOR_IMAGE` need to be set from day one.

Reply:
This was previously applied comment that in case users will have their own Deployment files, they are reponsible for changing the env variables (for Buildah etc.). Plus I replied on the previous comment you had https://github.com/strimzi/proposals/pull/170#discussion_r2347454160